pkg/kudoctl/cmd/upgrade.go

```diff
@@ -0,0 +1,165 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/cmd/install"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	upgradeExample = `
+		The upgrade argument must be a name of the package in the repository, a path to package in *.tar.gz format,
+		or a path to an unpacked package directory.
+
+		# Upgrade flink instance dev-flink to the latest version
+		kubectl kudo upgrade flink --instance dev-flink
+		
+		*Note*: should you have a local "flink" folder in the current directory it will take precedence over the remote repository.
+
+		# Upgrade flink to the version 1.1.1
+		kubectl kudo upgrade flink --instance dev-flink --version 1.1.1
+
+		# By default parameters are all reused from the previous installation, if you need to modify, use -p
+		kubectl kudo upgrade flink --instance dev-flink -p param=xxx`
+)
+
+type options struct {
+	InstanceName   string
+	Namespace      string
+	PackageVersion string
+	Parameters     map[string]string
+}
+
+// defaultOptions initializes the install command options to its defaults
+var defaultOptions = &options{
+	Namespace: "default",
+}
+
+// newUpgradeCmd creates the install command for the CLI
+func newUpgradeCmd() *cobra.Command {
+	options := defaultOptions
+	var parameters []string
+	upgradeCmd := &cobra.Command{
+		Use:     "upgrade <name>",
+		Short:   "Upgrade KUDO package.",
+		Long:    `Upgrade KUDO package from current version to new version.`,
+		Example: upgradeExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Prior to command execution we parse and validate passed parameters
+			var err error
+			options.Parameters, err = install.GetParameterMap(parameters)
+			if err != nil {
+				return errors.WithMessage(err, "could not parse parameters")
+			}
+			return runUpgrade(args, options)
+		},
+		SilenceUsage: true,
+	}
+
+	upgradeCmd.Flags().StringVar(&options.InstanceName, "instance", "", "The instance name.")
+	upgradeCmd.Flags().StringArrayVarP(&parameters, "parameter", "p", nil, "The parameter name and value separated by '='")
+	upgradeCmd.Flags().StringVar(&options.Namespace, "namespace", "default", "The namespace where the instance you want to upgrade is installed in. (default \"default\"")
```

Comment:
the default is provided by cobra... the result of this code is
`The namespace where the instance you want to upgrade is installed in. (default "default" (default "default")`

Reply:
why do you not use the `defaultOptions.Namespace` here for default?